Added ATA-over-Ethernet

KOLANICH opened the pull request.

generalmimon commented:
Thank you for starting to add test files to this repo.

Please add `README.md` files to each format directory (i.e. `hardware/aoe/README.md` for this PR) - this applies to all PRs even though I only write it here. Look at the existing READMEs to see how they should look like:

- second level (`## ...`) headings with sample file name,
- every test file should have a description, at least a short one - you can of course use Markdown to format text, link to other websites and so on,
- the **`Source: (...)` tag** declaring the origin of the file (most of the time a **URL** where the original file can be found, or `Source: own work` if the file was created by you, in which case I would appreciate a detailed description of how the file was created so the process can be easily reproduced if needed - the next bullet is related) - **this is a strict requirement** and **I insist on that**,
- if the file can be generated with a script or a CLI program, it's ideal when you include the exact commands/code used to create the file so that the creation is reproducible, format generation options can be easily tweaked and other variants of the file can be created.

---

A secondary issue reported by the CI check is the missing copyright and license information - I've decided to use [REUSE](https://reuse.software) in this repo, because I anticipate that a significant number of samples will not be our original works but taken from elsewhere, and I think it's fair to give credit to the authors (and most licenses, including MIT, legally require this, yet it is often overlooked and rarely enforced) and clearly state what license each file is under.

If you install the CLI `reuse` tool (https://github.com/fsfe/reuse-tool), you can use

* `reuse addheader` to add the copyright/licensing info in the machine-readable format,
* `reuse lint` to check the entire repo for REUSE issues and
* `reuse download --all` (**must be run at the repository root dir**, otherwise it creates a new `LICENSES/` folder in the current working directory - I forget to do this almost every time) to download the missing license texts to `LICENSES/` if `reuse lint` reports `Missing licenses:`. 

However, I can do this myself, provided that you properly add the `Source:` tags everywhere (but I don't know what name and e-mail would you prefer for your own works).

KOLANICH commented:
Thank you for the extended explaination.

1. About licenses. When a file is copyrighted by someone, like in scala_signature format, I have added the copyright files. Where they are not (BSON, CBOR and MsgPack were specially generated by a python script), I have added nothing.
2. I don't remember how exactly I have generated the files in this PR. I remember that I have taken some userspace ATA-over-Ethernet impl, bound it to a loopback interface, then connected to it using a kernel driver and used `smartctl` on the device. Then I have processed them with an own script (I guess it is not saved, I used it when I was developing the spec from the PR https://github.com/kaitai-io/kaitai_struct_formats/issues/208) that dumped pcap packets bodies into files and names them accordingly. There were about 100 files (tuat PR has an archive attached), but I guess I should not put the 100 files into this repo.
